File: backend/EcommerceApi/Models/pedido_model.cs
namespace EcommerceApi.Models;

public class Pedido
{
    public int Id { get; set; }
    public int UsuarioId { get; set; }
    public decimal Total { get; set; }
    public string Estado { get; set; } = "Pendiente"; // Pendiente, Pagado, Enviado, Entregado, Cancelado
    public string DireccionEnvio { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public DateTime? FechaEntrega { get; set; }
    
    // Relaciones
    public Usuario? Usuario { get; set; }
    public ICollection<PedidoItem> Items { get; set; } = new List<PedidoItem>();
}

File: frontend/ecommerce-app/src/app/shared/models/pedido.model.ts
import { Producto } from './producto.model';

export interface Pedido {
  id: number;
  usuarioId: number;
  usuarioNombre: string;
  total: number;
  estado: 'Pendiente' | 'Procesando' | 'Enviado' | 'Entregado' | 'Cancelado' | 'Pagado';
  direccionEnvio: string;
  fechaCreacion: string;
  fechaPago?: string;
  items: PedidoItem[];
  numeroSeguimiento?: string;
  servicioEnvio?: string;
  fechaDespacho?: string;
  fechaEntrega?: string;
  metodoPago?: string;
  transaccionId?: string;
}

export interface PedidoItem {
  id?: number;
  pedidoId?: number;
  productoId: number;
  productoNombre: string;
  productoImagen?: string;
  cantidad: number;
  precioUnitario: number;
  subtotal: number;
  producto?: Producto;
}

export interface CrearPedidoDto {
  direccionEnvio: string;
}

export interface ActualizarEstadoPedidoDto {
  estado: string;
}
